### DISCUSSION TITLE
Raft log compaction

### DISCUSSION THREAD

**Discussion by aderouineau** [rqlite/rqlite#1679] [General]
Does the raft log get compacted regularly?

Imagine:
* A 3-node quorum cluster for a large (10+ GB) database that gets decent amount of updates every day: many additions & deletions, but overall stays around 10 GB.
* An application cluster, where each application instance is paired (on the same host) with a read-only rqlite node so that it can do local queries on cached data. The cluster is managed by an auto-scaling group, such that a new node may be brought up as needed.

When an application node gets brought up, the rqlite read-only node starts from scratch. Will it have to get the entire log? Or will it be able to get a recent full snapshot and fewer log entries?

**Comment by otoolep** [comment]
Assuming I follow your question, it's get the latter. A node will send it a snapshot, and any extra logs. This is because after a while no node has the entire log. This is standard Raft stuff, known as Log Truncation.

**Reply by aderouineau** [reply]
OK, that's good to know. Thanks!
